code/game/objects/items/inspector.dm

```diff
@@ -106,18 +107,27 @@
 	if(!cell || !cell.use(INSPECTOR_ENERGY_USAGE_LOW))
 		balloon_alert(user, "check cell!")
 		return ITEM_INTERACT_BLOCKING
-	if(!isitem(interacting_with))
-		return ITEM_INTERACT_BLOCKING
-	var/obj/item/contraband_item = interacting_with
-	var/contraband_status = contraband_item.is_contraband()
-	if((!contraband_status && scans_correctly) || (contraband_status && !scans_correctly))
+
+	if(iscarbon(interacting_with)) //Prevents insta scanning people
+		if(!COOLDOWN_FINISHED(src, scanning_person))
+			return ITEM_INTERACT_BLOCKING
+
+		visible_message(span_warning("[user] starts scanning [interacting_with] with [src]"))
+		to_chat(interacting_with, span_userdanger("[user] is trying to scan you for contraband!"))
```

Comment:
balloon alerts instead

Reply:
why balloon alerts? Userdanger is big red text and it is used for example when you're getting stripped or attacked aka it is noticeable